xbmc/music/MusicUtils.cpp

```diff
@@ -718,27 +718,30 @@ bool IsItemPlayable(const CFileItem& item)
   if (item.IsPVR() || item.IsPlugin() || item.IsScript() || item.IsAddonsPath())
     return false;
 
-  // Check for right window
-  const int winID = CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow();
-  if (winID != WINDOW_MUSIC_NAV && winID != WINDOW_HOME)
-    return false;
-
-  if (item.m_bIsFolder)
+  if (item.m_bIsFolder &&
+      (item.IsMusicDb() || StringUtils::StartsWithNoCase(item.GetPath(), "library://music/")))
   {
     // Exclude top level nodes - eg can't play 'genres' just a specific genre etc
     const XFILE::MUSICDATABASEDIRECTORY::NODE_TYPE node =
         XFILE::CMusicDatabaseDirectory::GetDirectoryParentType(item.GetPath());
     if (node == XFILE::MUSICDATABASEDIRECTORY::NODE_TYPE_OVERVIEW)
       return false;
+
+    return true;
   }
 
-  if (item.HasMusicInfoTag() && item.CanQueue() && !item.IsParentFolder())
+  if (item.HasMusicInfoTag() && item.CanQueue())
     return true;
   else if (item.IsPlayList() && item.IsAudio())
     return true;
-  else if (!item.m_bIsShareOrDrive && item.m_bIsFolder && !item.IsParentFolder())
+  else if (item.IsAudio())
     return true;
-
+  else if (!item.m_bIsShareOrDrive && item.m_bIsFolder)
+  {
+    // Not a music-specific folder (like file:// or nfs://). Allow play if context is Music window.
```

Comment:
Looks like an hack. Isn't setting a hint property a better alternative?

Reply:
I do agree, but the logic was the same before this PR and honestly, I do not have an idea how such a hint should look like like and where this hint could be given.